# Part -2 Python Basics (Conditional Statements)

1) A company decided to give bonus of 5% to employee if his/her year of service is more than 5 years.
Ask user for their salary and year of service and print the net bonus amount.

In [12]:
#Type your code here
salary = float(input("Enter your salary: "))
years = int(input("Enter service years: "))

if years > 5:
    bonus = salary * 0.05
    print(f"Bonus amount is: {bonus} ")

else:
    print("No Bonus")    

Bonus amount is: 6000.0 



2) Write a program to check whether a person is eligible for voting or not. (accept age from user) if age is greater than 17 eligible otherwise not eligible

In [2]:
#Type your code here
age = int(input("Enter your age: "))

if age > 17:
    print("You are eligible for voting")

else:
    print("You are not eligible for voting")    

You are eligible for voting



3) Write a program to check whether a number entered by user is even or odd.

In [9]:
#Type your code here
number = int(input("Enter a number: "))

if number % 2 == 0:
    print("Number is Even")

else:
    print("Number is Odd")    

Number is Even


4) Write a program to check whether a number is divisible by 7 or not.
Show Answer

In [17]:
#Type your code here
num = int(input("Enter a number: "))

if num % 7 == 0:
    print("Number is divisible by 7")

else:
    print("Number is not divisible by 7")    

Number is divisible by 7


5) Write a program to display 
"Hello" if a number entered by user is a multiple of five , otherwise print "Bye".

In [10]:
#Type your code here
number = int(input("Enter a number: "))

if number % 5 == 0 :
    print("Hello")

else:
    print("Bye")    

Hello


7) Write a program to display the last digit of a number.


In [15]:
#Type your code here
number = int(input("Enter a number: "))

last_digit = number % 10
print(f"Last digit is: {last_digit}")

Last digit is: 4


9) Take values of length and breadth of a rectangle from user and print if it is square or rectangle.

In [17]:
#Type your code here 
length = int(input("Enter a length: "))
breadth = int(input("Enter a breadth: "))

if length == breadth:
    print("It is a square")

else:
    print("It is a rectangle")    

It is a rectangle


10) Take two int values from user and print greatest among them.

In [34]:
# Type your code here
num1 = int(input("Enter a number1: "))
num2 = int(input("Enter a number2: "))

if num1 > num2:
    print(f"Num1 is greatest: {num1}")

elif num2 > num1:
    print(f"Num2 is greatest: {num2}")    

else:
    print("Both are equal")    

Both are equal


11) A shop will give discount of 10% if the cost of purchased quantity is more than 1000.
Ask user for quantity
Suppose, one unit will cost 100.
Judge and print total cost for user.

In [38]:
#Type your code here
quantity = int(input("Enter quantity: "))
cost = quantity * 100

if cost > 1000:
    discount = cost * 0.10
    cost = cost - discount
    print("Discount is",discount)
    print(f"Total cost is: {cost}")

else:
    print("No discount")    

Discount is 150.0
Total cost is: 1350.0


12) A school has following rules for grading system:

a. Below 25 - F

b. 25 to 45 - E

c. 45 to 50 - D

d. 50 to 60 - C

e. 60 to 80 - B

f. Above 80 - A

Ask user to enter marks and print the corresponding grade.

In [24]:
#Type your code here
marks = float(input("Enter your marks: "))

if marks > 80:
    print("Grade A")

elif marks > 60 and marks <= 80 :
    print("Grade B")

elif marks > 50 and marks <= 60:
    print("Grade C")

elif marks > 45 and marks <= 50:
    print("Grade D")

elif marks > 25 and marks <= 45:
    print("Grade E")

else:
    print("Grade Fail")                 

Grade D


14)A student will not be allowed to sit in exam if his/her attendence is less than 75%.

Take following input from user

- Number of classes held

- Number of classes attended.

And print

- percentage of class attended

- Is student is allowed to sit in exam or not.


In [45]:
#Type your code here
held = int(input("Enter Number of classes held: "))
attended = int(input("Enter Number of classes attended: "))

percentage = (attended / held) * 100
print(f"Attendence Percentage: {round(percentage)}")

if percentage >= 75:
    print("Allowed to sit in exam")

else:
    print("Not Allowed")    

Attendence Percentage: 58
Not Allowed


15) Modify the above question to allow student to sit if he/she has medical cause. Ask user if he/she has medical cause or not ( 'Y' or 'N' ) and print accordingly.

In [47]:
#Type your code here
held = int(input("Enter Number of classes held: "))
attended = int(input("Enter Number of classes attended: "))
medical = input("Enter medical cause (Y/N): ")

percentage = (attended / held) * 100
print(f"Attendence Percentage: {round(percentage)}")

if percentage >= 75 or medical == "Y":
    print("Allowed to sit in exam")

else:
    print("Not Allowed")  

Attendence Percentage: 75
Allowed to sit in exam


16) Write a program to check if a year is leap year or not.

If a year is divisible by 4 then it is leap year but if the year is century year like 2000, 1900, 2100 then it must be divisible by 400.

In [ ]:
#Type your code here
year = int(input("Enter year: "))

if year % 400 == 0:
    print("Leap year")

elif year % 4 == 0 and year % 100 != 0:
    print("Leap year")

else:
    print("Not a Leap year")       

Leap year


17) Ask user to enter age, gender ( M or F ), marital status ( Y or N ) and then using following rules print their place of service.

if employee is female, then she will work only in urban areas.

if employee is a male and age is in between 20 to 40 then he may work in anywhere

if employee is male and age is in between 40 t0 60 then he will work in urban areas only.

And any other input of age should print "ERROR"

In [52]:
#Type your code here
age = int(input("Enter age: "))
gender = input("Enter gender (M/F): ")
marital = input("Enter Marital Status (Y/N): ")

if gender == "F":
    print("She will work in urban areas")

elif gender == "M" and age >= 20 and age <= 40:
    print("He may work in anywhere")

elif gender == "M" and age > 40 and age <= 60:
    print("He will work in urban areas")

else:
    print("Error")            

Error


6) Write a program to calculate the electricity bill (accept number of unit from user) according to the following criteria :
     Unit                                                     Price  
uptp 100 units                                             no charge
Next 200 units                                              Rs 5 per unit
After 200 units                                             Rs 10 per unit
(For example if input unit is 350 than total bill amount is Rs.3500
(For example if input unit is 97 than total bill amount is Rs.0
(For example if input unit is 150 than total bill amount is Rs.750

In [57]:
#Type your code here
units = int(input("Enter units: "))
bill = 0

if units <= 100:
    bill = 0

elif units <= 300:
    bill = (units - 100) * 5

else:
    bill = (200 * 5) + (units - 300) * 10

print("Total bill is Rs.", bill)

Total bill is Rs. 1500


13) Take input of age of 3 people by user and determine oldest and youngest among them.

In [54]:
#Type your code here
a = int(input("Enter age of first person: "))
b = int(input("Enter age of second person: "))
c = int(input("Enter age of third person: "))

#Oldest
if a >= b and a >= c:
    oldest = a

elif b >= a and b >= c:
    oldest = b

else:
    oldest = c

#Youngest
if a <= b and a <= c:
    youngest = a

elif b <= a and b <= c:
    youngest = b

else:
    youngest = c

print(f"Oldest is: {oldest}")
print(f"Youngest is: {youngest}")                        

Oldest is: 35
Youngest is: 25
